add MITx program report to staging and intermediate

rachellougee opened the pull request:
# What are the relevant tickets?
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Closes # --->
<!--- Fixes # --->
<!--- N/A --->
part of https://github.com/mitodl/ol-data-platform/issues/852

# Description (What does it do?)
<!--- Describe your changes in detail -->
Adds `stg__edxorg__s3__program_learner_report` and `int__edxorg__mitx_program_certificates`, which are deduped from `raw__edxorg__program_learner_report`

It doesn't replace the MicroMasters program data in `__micromasters_program_certificates_non_dedp` yet as we need to figure out the discrepancies first


# How can this be tested?
<!---
Please describe in detail how your changes have been tested.
Include details of your testing environment, any set-up required
(e.g. data entry required for validation) and the tests you ran to
see how your change affects other areas of the code, etc.
Please also include instructions for how your reviewer can validate your changes.
--->
Needs to update [this line](https://github.com/mitodl/ol-data-platform/blob/main/src/ol_dbt/profiles.yml#L46) to `mitol-ol-data-lake-production.trino.galaxy.starburst.io` to run against `dev_production` due to size of this table
```
dbt build --select stg__edxorg__s3__program_learner_report
dbt build --select int__edxorg__mitx_program_certificates
```

rachellougee commented on src/ol_dbt/models/staging/edxorg/_edxorg_sources.yml:
```diff
@@ -451,3 +451,78 @@ sources:
       description: timestamp, specifying when email opt in preference was most recently
         updated
+
+  - name: raw__edxorg__program_learner_report
```
@pdpinch Can you review these field definitions to see if it's what you expected?

pdpinch commented:
I looked through my inbox and found a PDF that documents the field definitions. Let me know if this helps, or just raises more questions.
 
[Program Console - Program Data Reports Guide.pdf](https://github.com/mitodl/ol-data-platform/files/12929374/Program.Console.-.Program.Data.Reports.Guide.2.pdf)

rachellougee commented:
@pdpinch Perfect. That explains all the fields definition.

abeglova commented on src/ol_dbt/models/staging/edxorg/stg__edxorg__s3__program_learner_report.sql:
```diff
@@ -0,0 +1,86 @@
+        , "course title" as course_title
+        , track as courserunenrollment_enrollment_mode
+        , cast("user id" as integer) as user_id
+        , cast(completed as boolean) as completed_course
```
Should this be course_is_completed to match our normal naming scheme for booleans?

abeglova replied:
Same for other booleans in this table

abeglova commented on src/ol_dbt/models/staging/edxorg/stg__edxorg__s3__program_learner_report.sql:
```diff
@@ -0,0 +1,86 @@
+        , cast("purchased as bundle" as boolean) as purchased_as_bundle
+        , case
+            when "program title" = 'Finance MIT' and "program uuid" like '%56966c7' then 'Finance' --noqa: CV10
+            when "program title" = 'Finance' and "program uuid" like '%e11a748' then 'MIT Finance' --noqa: CV10
```
Is there a typo? You are changing 'Finance MIT' to 'Finance'
And 'Finance' to 'MIT Finance' . Should both be changed to  'MIT Finance'?

abeglova replied:
Or is the issue that you are trying to make sure the program titles are different for different program uuids?

rachellougee replied:
No, its not typo. As you can see, Finance comes with 3 different names and 2 different UUID. My understanding is that  `c22ff74a-6bec-4357-b941-d3eaf56966c7` is MIT Finance (new version), `50e72d85dfc7402ba3173ff51e11a748` is Finance. So this ensures title are consistent with UUID.
![Pasted Graphic](https://github.com/mitodl/ol-data-platform/assets/3138890/b5e0eafe-5e26-4ca9-8057-10b079b96bdd)

abeglova replied:
So maybe you can remove the `"program title" =` part of the line and just keep the uuid part? If the title is already correct already that's  fine

abeglova replied:
Just so it's clearer that you're assigning titles based on uuid

rachellougee replied:
pre-commit didn't like when I commit the whole UUID, so I used the partial match but didn't want to run into collision with partial uuid. But now I think about it, it should be fine to just keep the uuid part. I updated it

abeglova commented on src/ol_dbt/models/staging/edxorg/stg__edxorg__s3__program_learner_report.sql:
```diff
@@ -0,0 +1,86 @@
+        , case
+            when "date completed" = 'null' then null --noqa: CV10
+            else to_iso8601(from_iso8601_timestamp(
+                regexp_replace("date completed", '(\d{4}-\d{2}-\d{2})[ ](\d{2}:\d{2}:\d{2})[ ]Z', '$1T$2Z') --noqa: CV10,LT05
```
Can you use `date_parse(` here instead of `regexp_replace(`?

rachellougee replied:
done, its actually much shorter to use date_parse. Thanks

abeglova reviewed (changes requested):
Back to you with some comments. Also the refreshing stg__edxorg__s3__program_learner_report has been running for about 30 min now, even though I changed the host on dev_production to  host: mitol-ol-data-lake-qa-0.trino.galaxy.starburst.io. Not sure if that's expected

rachellougee commented:
Per discussion, this raw table is very large so it needs to run under `mitol-ol-data-lake-production.trino.galaxy.starburst.io`. It should be less than 5 mins. 
@abeglova Addressed your comments. Please take a look when you get a chance.

abeglova commented on src/ol_dbt/models/staging/edxorg/stg__edxorg__s3__program_learner_report.sql:
```diff
@@ -0,0 +1,86 @@
+        , cast("currently enrolled" as boolean) as courserunenrollment_is_active
+        , cast("purchased as bundle" as boolean) as purchased_as_bundle
+        , case
+            when "program title" = 'Finance MIT' and "program uuid" like '%56966c7' then 'Finance' --noqa: CV10
```
Can you update .pre-commit-config.yaml to ignore CV10? the syntax should be 
```
[sqlfluff]
exclude_rules = CV10
```
https://docs.sqlfluff.com/en/stable/configuration.html

abeglova reviewed (approved):
LGTM - although i think you should just disable CV10 instead of having so may noqa comments
